### ISSUE TITLE
errors at run start and finish

### ISSUE THREAD

**Issue by wakamex** [RainbowMiner/RainbowMiner#1229]
could be related to recent change in paths?

logs attached [debug_2020-11-05.zip](https://github.com/RainbowMiner/RainbowMiner/files/5495741/debug_2020-11-05.zip)

```
New-Object : The value supplied is not valid, or the property is read-only. Change the value, and then try again.
At D:\rainbowminer\Modules\Core.psm1:2052 char:28
+             $ActiveMiner = New-Object $Miner.API -Property @{
+                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidData: (:) [New-Object], Exception
+ FullyQualifiedErrorId : SetValueException,Microsoft.PowerShell.Commands.NewObjectCommand

Starting mining operation ..
Method invocation failed because [System.Collections.ObjectModel.Collection`1[[System.Management.Automation.PSObject, System.Management.Automation, Version=6.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]] does not contain a method named 'GetStatus'.
At D:\rainbowminer\Modules\Core.psm1:2261 char:13
+         if ($_.GetStatus() -eq [MinerStatus]::RunningFailed) {
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [], RuntimeException
+ FullyQualifiedErrorId : MethodNotFound

Method invocation failed because [System.Collections.ObjectModel.Collection`1[[System.Management.Automation.PSObject, System.Management.Automation, Version=6.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]] does not contain a method named 'GetProcessIds'.
At D:\rainbowminer\Modules\Core.psm1:2300 char:68
+ ... = @($Global:ActiveMiners | Foreach-Object {$_.GetProcessIds()} | Wher ...
+                                                ~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [], RuntimeException
+ FullyQualifiedErrorId : MethodNotFound

Split-Path : Cannot bind argument to parameter 'Path' because it is null.
At D:\rainbowminer\Modules\Core.psm1:2301 char:85
+ ... lobal:ActiveMiners | Foreach-Object {Split-Path -Leaf $_.Path} | Sele ...
+                                                           ~~~~~~~
+ CategoryInfo          : InvalidData: (:) [Split-Path], ParameterBindingValidationException
+ FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SplitPathCommand
```

**Comment by wakamex** [comment]
similar errors with a brand new install, with only the config folder copied over

**Comment by wakamex** [comment]
it was just successfully mining toward a new rental on MRR then at the end of a run loop it errored and temporarily disabled MRR...

**Comment by RainbowMiner** [comment]
I check asap. Sorry for that! My test rigs all run fine.

**Comment by RainbowMiner** [comment]
Have you manually set CPUMiningAffinity and CPUMiningThreads in config.txt?

**Comment by RainbowMiner** [comment]
For a quick fix, please manually unzip the release v4.6.4.1 into your miner folder. That should cure it for the rental and give me some time to find out, what is going on.

**Comment by wakamex** [comment]
I did have CPUMiningThreads set to 30, hoping that would cause most miners to prefer more than 16 threads, which didn't seem to be the case. unsetting it didn't seem to fix things.

v.4..6.4.0 seems to be running well. thanks for pointing out the rollback option.

caught the first set of [runtime errors i kept getting](https://justpaste.it/1p367)

**Comment by RainbowMiner** [comment]
With the new release, you will be able to use more threads. Actually, that new feature might be the cause for the trouble. I am checking, currently. 
I have tried the new release on a 2 x AMD Epyc system, together 64 cores, 128 threads. Xmrig has been able to address all of them.

**Comment by RainbowMiner** [comment]
This bug should now be fixed, with v4.6.4.3
Problem was the CPUMiningAffinity - I have extended the affinity calculations to a maximum of 1024 threads and run into a problem to check for zero with bigint numbers. Sorry for the errors!

**Comment by wakamex** [comment]
no errors with the new version, great quick turnaround :)
